### PR TITLE
Remove the last bits of biggus

### LINKED ISSUE #2443: Remove all biggus imports from Iris codebase
```
ldreyer@eld347 $ grep -ir --color "import biggus"
coords.py:import biggus
fileformats/grib/message.py:import biggus
fileformats/grib/__init__.py:import biggus
fileformats/netcdf.py:import biggus
tests/unit/concatenate/test_concatenate.py:import biggus
tests/unit/cube/test_Cube__operators.py:import biggus
cube.py:import biggus
```

### PR DESCRIPTION
**Pull request by dkillick**
Remove the final biggus imports (and usages) from the codebase*.

In essence, this means:

 * delete a couple of spurious (and unused!) biggus import statements,
 * sort out some test modules, and
 * tidy up the lazy data API where the tests very pleasingly showed up a couple of gaps in it.

Fixes #2443.
&nbsp;

\* Well, not _quite_: I explicitly **haven't** dealt with grib here - we'll [do](https://github.com/SciTools/iris/issues/2342) that [later](https://github.com/SciTools/iris/issues/2306).

### CODE REVIEW

File: lib/iris/cube.py [hunk 1]
```diff
@@ -1738,8 +1737,7 @@ def data(self):
                 result = as_concrete_data(self._dask_array,
                                           nans_replacement=ma.masked,
                                           result_dtype=self.dtype)
-                self._numpy_array = result
-                self.dtype = None
+                self.data = result
```
Comment (dkillick) [inline lib/iris/cube.py]: Changing the instance's `data` attribute via the data setter ensures all the appropriate processing for the new data value is applied when the instance's lazy data is computed.

File: lib/iris/_lazy_data.py [hunk 2]
```diff
@@ -97,8 +97,11 @@ def as_concrete_data(data, **kwargs):
 
     """
     if is_lazy_data(data):
-        # Realise dask array.
-        data = data.compute()
+        # Realise dask array, ensuring the data result is always a NumPy array.
+        # In some cases dask may return a scalar numpy.int/numpy.float object
+        # rather than a numpy.ndarray object.
+        # Recorded in https://github.com/dask/dask/issues/2111.
+        data = np.asanyarray(data.compute())
```
Comment (bjlittle) [inline lib/iris/_lazy_data.py]: @dkillick Nice. Kinda glad we centralized this, huh ... :wink:

File: lib/iris/cube.py (hunk 1 above)
Comment (bjlittle) [inline lib/iris/cube.py]: Oh my ... this is interesting. I agree in the approach, but that'll now result in **both** the `dtype` and in addition the `fill_value` being cleared. It'll be interesting to see if this causes any of the existing tests to fail. I suspect something might barf.

Also, it means that we should join the dots here, and make sure that the `Cube.__init__` is setting the `data` by using the `self.data` setter ... and it's currently not! See [here](https://github.com/SciTools/iris/pull/2451/files#diff-d1f30240d0d146f09f58577e718ba6a6R719)

File: lib/iris/tests/unit/cube/test_Cube__operators.py [hunk 3]
```diff
@@ -19,173 +19,247 @@
...
+
+import dask.array as da
 import numpy as np
 import numpy.ma as ma
-import biggus
-from biggus._init import _Elementwise
+
+import iris
+from iris.coords import DimCoord
+from iris._lazy_data import as_lazy_data
```
Comment (bjlittle) [inline lib/iris/tests/unit/cube/test_Cube__operators.py]: @dkillick Import order, `iris._lazy_data` sorts before `iris.coords` ...

File: lib/iris/_lazy_data.py (hunk 2 above)
Comment (dkillick) [inline lib/iris/_lazy_data.py]: @bjlittle oh so very glad!

File: lib/iris/tests/unit/cube/test_Cube__operators.py (hunk 3 above)
Comment (dkillick) [inline lib/iris/tests/unit/cube/test_Cube__operators.py]: Whoops!

**Comment by dkillick** [comment]
I've pulled out the changes to the cube data API as they were getting too big for this PR. I'll raise them separately presently and reference that PR here when it is ready.

**Comment by dkillick** [comment]
We also have maths problems... this is fixed by #2458

**Comment by dkillick** [comment]
Review actions and a rebase now that the math changes (#2458) have been merged.

**Comment by bjlittle** [comment]
@dkillick Great, thanks :+1:

**Comment by dkillick** [comment]
Thanks @bjlittle!

**Pull request merged**